db/migrate/20191109194244_add_user_to_pre_auth_states.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserToPreAuthStates < ActiveRecord::Migration[5.2]
+  def change
+    add_reference :pre_auth_states, :user, index: true
```

Comment:
I think you need to do something here about existing PreAuthStates otherwise you'll have a bunch of invalid records in the DB (since the belongs_to :user is now required).

Reply:
I think the changes I made to the `PreAuthState`s are not necessary anymore so I plan to get rid of them.

Reply:
Deleted this migration.